recipes/nsqd.rb

```diff
@@ -25,24 +25,89 @@
 end
 
 if node['nsq']['setup_services']
-  template '/etc/init/nsqd.conf' do
-    action :create
-    source 'upstart.nsqd.conf.erb'
-    mode '0644'
-    # need to stop/start in order to reload config
-    if node['nsq']['reload_services']
-      notifies :stop, 'service[nsqd]', :immediately
-      notifies :start, 'service[nsqd]', :immediately
+
+  if node.platform_family?('debian') && node['platform_version'].to_f >= 18.04
+
+    standard_error = node['nsq']['nsqd']['standard_error'].split('file:/')[1]
+    standard_output = node['nsq']['nsqd']['standard_output'].split('file:/')[1]
+
+    directory '/var/log/nsqd' do
+      action :create
+      mode '0770'
+      owner 'nsqd'
+      group 'nsqd'
+      recursive true
+    end
+
+    file standard_output do
+      action :create
+      mode '0770'
+      owner 'nsqd'
+      group 'nsqd'
+    end
+
+     file standard_error do
+      action :create
+      mode '0770'
+      owner 'nsqd'
+      group 'nsqd'
+    end
+
+
+    template '/srv/nsqd-start.sh' do
+      action :create
+      source 'nsqd-start.sh.erb'
+      mode '0550'
+      # mode '0777'
+      owner 'nsqd'
+      group 'nsqd'
+    end
+
+    template '/etc/systemd/system/nsqd.service' do
+      action :create
+      source 'systemd.nsqd.conf.erb'
+      mode '0644'
+      notifies :run, 'execute[systemctl-daemon-reload]'
+      # need to stop/start in order to reload config
+      if node['nsq']['reload_services']
+        notifies :stop, 'service[nsqd]', :immediately
+        notifies :start, 'service[nsqd]', :immediately
+      end
+    end
+
+    execute 'systemctl-daemon-reload' do
+      command 'systemctl daemon-reload'
+      action :nothing
+    end
+  else
+    template '/etc/init/nsqd.conf' do
+      action :create
+      source 'upstart.nsqd.conf.erb'
+      mode '0644'
+      # need to stop/start in order to reload config
+      if node['nsq']['reload_services']
+        notifies :stop, 'service[nsqd]', :immediately
+        notifies :start, 'service[nsqd]', :immediately
+      end
     end
   end
 
   service 'nsqd' do
-    provider Chef::Provider::Service::Upstart
-    action [:enable, :start]
-    supports stop: true, start: true, restart: true, status: true
-    # Conditionally subscribe to version updates
-    if node['nsq']['reload_services']
-      subscribes :restart, "ark[#{nsq_release}]", :delayed
+    if node.platform_family?('debian') && node['platform_version'].to_f >= 18.04
```

Comment:
I'd suggest doing `node.platform?('ubuntu')` here, since you're then matching the version based on its ubuntu version and not its debian version.